SilKit/include/silkit/util/serdes/Serializer.hpp

```diff
@@ -151,13 +153,26 @@ class Serializer
     /*! \brief Serializes the end of an optional value. */
     void EndOptional() {}
 
-    void BeginUnion(int)
+    /*! \brief Serialize the start of a union with a particular discriminator.
+     * Only the active union member must be serialized prior to calling EndUnion.
+     * \param discriminator 1-based index of the active union member
+     */
+    void BeginUnion(const int discriminator)
     {
-        throw SilKitError("Unions are currently not supported.");
+#ifndef NDEBUG
+        if (discriminator < 0)
+            throw SilKitError{"Union discriminator is negative"};
+        if (discriminator == 0)
```

Comment:
Wouldn't this be allowed according to the "spec" in COE-66476? I.e. write the value 0 for the tag to the serialize buffer and then nothing for the actual union value?

Reply:
Good point. I'll experiment some more on the internal side to see if a union value can be 'cleared' somehow.

Reply:
And after some further investigation, it turns out that it is actually possible to clear the union explicitly in some (but not all) language bindings on the internal side.

Thank you for the feedback, the check and throw has been removed!